### DISCUSSION TITLE
Setting element frame line width in the context of focus?

### DISCUSSION THREAD

**Discussion by obleck1** [ImpulseAdventure/GUIslice#541] [Q&A]
Hello, 

Thanks in advance to everyone that put this library together, and the Builder as well.  It's phenomenal work. I'm not a programming novice, but getting into the weeds is a bit challenging for me, so thanks in advance on this.

I got lib running in a Teensy 4.0 with a non-touch ILI8941 (? forget offhand) screen, with three physical buttons/pin inputs, and it's working fine.  I generated the code using the Builder and some customizations from various examples and it will happily select elements in a page.

However, the line width of the frame it generates around the selected/focused element looks to be about 1 pixel wide and is barely discernable.  

Question: how can I set the frame line width of an element in focus to something larger on a global scale?  

I searched the doc's pretty exhaustively, and it seems one can set the frame color of deselected, selected, and edit mode, with gslc_SetFocusCol(...), but there I can't find an equivalent for the frame's line width.

(Also, there doesn't seem to be a way to set these colors either in Builder for the "focuser", just for each element separately, though they are getting set somewhere.)

I imagine for a rectangle say, when in focus, it's drawing a rectangle with a frame using perhaps gslc_DrawFrameRect(...), which is adding some frame to it.  Is this frame set-able? Perhaps there is some similar parameter set for the corner radiusing when using rounded elements?

Any insights would be greatly appreciated.

Thanks

**Comment by obleck1** [comment]
Note: I haven't been able to locate the src that generates the focus functionality, but I imagine when some element is focused on, it's drawn at a slightly larger size with a fill color that will become the frame color, and then overdrawn, centered, by the original element.  Or something like that.  

If so, is there some parameter the code uses in determining the increased size of the first element drawn?  That would determine the frame line width/2.

**Comment by Pconti31** [comment]
@obleck1 Sorry, but the API has no such function.  I believe your idea above is the best attempt but I I'm not sure where you would add this feature.  
However, here is a flow chart I made sometime back.  Maybe it will help you look through the code.

![05-03-trackInputDiagram](https://github.com/ImpulseAdventure/GUIslice/assets/44075106/c758c622-c72f-4435-a851-31ea45aa016e)

Paul--
